docs/DeveloperGuide.md

```diff
@@ -149,103 +149,104 @@ Classes used by multiple components are in the `seedu.storemando.commons` packag
-
-![UndoRedoState0](images/UndoRedoState0.png)
-
-Step 2. The user executes `delete 5` command to delete the 5th item in the inventory. The `delete` command
-calls `Model#commitStoreMando()`, causing the modified state of the inventory after the `delete 5` command executes to
-be saved in the `storeMandoStateList`, and the `currentStatePointer` is shifted to the newly inserted inventory state.
-
-![UndoRedoState1](images/UndoRedoState1.png)
+#### Reminder Feature
+
+##### Implementation
+
+This portion describes the implementation of the reminder feature which allows users to view items that are expiring 
+a certain number of days which will be given by the user in the command.
+
+1. When the user keys in a command string, `execute` command of the `LogicManager` is called with the given string as input. 
+2. In the method, `LogicManager` calls on the `parseCommand` method of StoreMandoParser to parse the given command.
+3. `StoreMandoParser` parses the string and determines the command given is a `ReminderCommand`. 
+    Then, a `ReminderCommandParser` object is created to further parse the command.
+4. `StoreMandoParser` then calls on the `parse` method of `ReminderCommandParser` to parse the arguments provided.
+5. `ReminderCommandParser` calls on its own timeConversion method to convert the string into an integer. A 
+    `CommandParseException` will be thrown if this is not possible. 
+6. `ReminderCommandParser` then calls on the constructor of `ItemExpiringPredicate` with the integer as parameter to 
+    create an `ItemExpiringPredicate` object and then calls on the constructor of `ReminderCommand` with the `ItemExpiringPredicate` object as 
+    a parameter. 
+7. The `ReminderCommand` object will be created and returned to `StoreMandoParser` which returns it to `LogicManager` 
+    which then calls on the `execute` method of `ReminderCommand` with a `model` as argument.
+9. `ReminderCommand` calls the `updateCurrentPredicate` method of `model` and passes its own `ItemExpiringPredicate`
+    as argument.
+10. `ReminderCommand` calls the `getCurrentPredicate` method of `model` to obtain the current predicate and uses it
+    to update the list by calling on `updateFilteredItemList` method of `model` with the current predicate as argument.
+11. `ReminderCommand` then creates a `ItemComparatorByExpiryDate` object and calls `model`'s `updateSortedList` with 
+    `ItemComparatorByExpiryDate` as argument to sort the list.
+12. Finally, a `CommandResult` object is created and returned to `LogicManager`.    
+    
 
-Step 3. The user executes `add n/David …​` to add a new item. The `add` command also calls `Model#commitStoreMando()`,
-causing another modified inventory state to be saved into the `storeMandoStateList`.
+The following sequence diagram shows how the reminder feature works:
 
-![UndoRedoState2](images/UndoRedoState2.png)
+![ReminderSequenceDiagram](images/ReminderWeeksSequenceDiagram.png)
```

Comment:
I think it would be better if we include the sequence diagram before the implementation and use the sequence diagram to explain the implementation instead. It would be easier for the reader to understand the implementation.

Reply:
Yup that is the format we agreed on. 
Intro to the feature
Sequence diagram
Steps of implementation
Activity diagram
Design consideration